examples/with-mobx-state-tree-typescript/package.json

```diff
@@ -9,6 +9,7 @@
   "dependencies": {
     "mobx": "^5.9.0",
     "mobx-react": "^5.4.3",
```

Comment:
Can we drop this dep?

Reply:
Currently mobx-react is still used as the <Provider>, so no. But yes if we can modify `store.ts` to useContext in a pattern similar to this:

````
const RootStoreContext = createContext<null | RootInstance>(null);

export const Provider = RootStoreContext.Provider;
export function useMst() {
  const store = useContext(RootStoreContext);
  if (store === null) {
    throw new Error("Store cannot be null, please add a context provider");
  }
  return store;
}
````

But I couldn't get that to work. Would love to if someone knows how to swap that out. 

@impulse did this nicely in this create-react-app example: https://github.com/impulse/react-hooks-mobx-state-tree/blob/master/src/models/Root.ts